tests/integration/cmd_dev_test.go

```diff
@@ -1397,23 +1401,6 @@ var _ = Describe("odo dev command tests", func() {
 		})
 	})
 
-	When("running odo dev and build command throws an error", func() {
```

Comment:
Why was this test removed? It still makes sense, no? By the way, while testing this, I noticed that if the build command fails to run immediately, `odo dev` keeps logging forever while trying to run the build command (until the command is fixed in the Devfile).

```
❯ odo dev                                                                                                                                                                                                                                      
  __                                                                                                                                                                                                                                           
 /  \__     Developing using the my-nodesjs-with-k8s-uri-with-var Devfile                                                                                                                                                                      
 \__/  \    Namespace: default                                                                                                                                                                                                                 
 /  \__/    odo version: v3.0.0-beta1                                                                                                                                                                                                          
 \__/                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
↪ Deploying to the cluster in developer mode                                                                                                                                                                                                   
 •  Waiting for Kubernetes resources  ...                                                                                                                                                                                                      
 ⚠  Pod is Pending                                                                                                                                                                                                                             
 ✓  Pod is Running                                                                                                                                                                                                                             
 ✓  Syncing files into the container [208ms]                                                                                                                                                                                                   
 ✗  Building your application in container on cluster (command: install) [2s]                                                                                                                                                                  
                                                                                                                                                                                                                                               
up to date, audited 33 packages in 2s                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
3 packages are looking for funding                                                                                                                                                                                                             
  run `npm fund` for details                                                                                                                                                                                                                   
                                                                                                                                                                                                                                               
found 0 vulnerabilities                                                                                                                                                                                                                        
Error occurred on Push - watch command was unable to push component: unable to exec command [/bin/sh -c cd $PROJECT_SOURCE && (npm install && exit 10) 1>>/proc/1/fd/1 2>>/proc/1/fd/2]: error while streaming command: command terminated with
 exit code 10                                                                                                                                                                                                                                  
                                                                                                                                                                                                                                               
 ✓  Syncing files into the container [210ms]                                                                                                                                                                                                   
 ✗  Building your application in container on cluster (command: install) [1s]   
                                                                                                                                                                                                                                               
up to date, audited 33 packages in 2s                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
3 packages are looking for funding                                                                                                                                                                                                             
  run `npm fund` for details                                                                                                                                                                                                                   
                                                                                                                                                                                                                                               
found 0 vulnerabilities                                                                                                                                                                                                                        
                                                                                                                                                                                                                                               
up to date, audited 33 packages in 1s                                                                                                                                                                                                          
                                                                                                                                                                                                                                               
3 packages are looking for funding                                                                                                                                                                                                             
  run `npm fund` for details                                                                                                                                                                                                                   
                                                                                                                                                                                                                                               
found 0 vulnerabilities                                    
Error occurred on Push - watch command was unable to push component: unable to exec command [/bin/sh -c cd $PROJECT_SOURCE && (npm install && exit 10) 1>>/proc/1/fd/1 2>>/proc/1/fd/2]: error while streaming command: command terminated with
 exit code 10                                              

 ✓  Syncing files into the container [208ms]
 ✗  Building your application in container on cluster (command: install) [1s]

up to date, audited 33 packages in 2s

3 packages are looking for funding
  run `npm fund` for details                               

found 0 vulnerabilities                                    

up to date, audited 33 packages in 1s

3 packages are looking for funding
  run `npm fund` for details                               

found 0 vulnerabilities               
```

Reply:
> > Curious as to why this test was removed
> 
> I cannot find a way to test it with the current behaviour.
> 
> > By the way, while testing this, I noticed that if the build command fails to run immediately, `odo dev` keeps logging forever while trying to run the build command again and again (until the command is fixed in the Devfile).
> 
> What would be the correct behavior, in your opinion? The tests were checking that the command `odo dev` were exiting as soon as the build command failed. Do we still want to exit, or wait for a change to the sources?

Whoops, I just saw your comment.
I remember I found it odd that we currently (in the `main` branch) exit if the (optional) build command fails, but keep watching for changes if the run command fails. IMO, it would make more sense (and also be more consistent) not to exit but keep watching for changes if the build command fails.
@kadel Any thoughts on this?

Because, in this PR, if I purposely run `odo dev --no-watch` with a failing build command, `odo` would keep flooding the terminal by trying to run the build command, which would quickly become annoying, I think.

Reply:
I'll see if it is possible to stop immediately in no-watch mode when build fails

Reply:
I've added a commit to try and fix the problem, and added back the test, with --no-watch flag